src/main/java/org/jabref/model/metadata/MetaData.java

```diff
@@ -141,7 +141,6 @@ public void setCiteKeyPattern(List<String> defaultValue, Map<EntryType, List<Str
 
     public void setSaveActions(FieldFormatterCleanups saveActions) {
         this.saveActions = Objects.requireNonNull(saveActions);
-        postChange();
```

Comment:
Removing `postChange()` here seems to disable the ability to store SaveActions in the BibFile completly. I believe the real problem is in `LibraryPropertiesDialogViewModel::setSaveActions` (line 131): `Cleanups.DEFAULT_SAVE_ACTIONS` are stored in the `initialMetaData` variable which refs to the actual library instaed to the ViewModel.